### PR TITLE
CLI: Support transfer with seed

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

We aren't the only ones who've managed to send funds to derived addresses

#### Summary of Changes

Add hidden support for `SystemInstruction::TransferWithSeed` to CLI

Works on devnet, but needs tests fixed up + added

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -1853,6 +1888,17 @@ where
     }
 }
 
+fn derived_address_seed_validator(s: String) -> Result<(), String> {
```
Comment (CriesofCarrots) [inline cli/src/cli.rs:1891]: Should we move this into clap-utils/src/input_validators.rs?

File: cli/src/cli.rs [hunk 2]
```diff
@@ -1108,17 +1116,40 @@ fn process_transfer(
...
     let from = config.signers[from];
+    let mut from_pubkey = from.pubkey();
 
     let (recent_blockhash, fee_calculator) =
         blockhash_query.get_blockhash_and_fee_calculator(rpc_client, config.commitment)?;
 
     let nonce_authority = config.signers[nonce_authority];
     let fee_payer = config.signers[fee_payer];
 
+    let derived_parts = derived_address_seed.zip(derived_address_program_id);
```
Comment (CriesofCarrots) [inline cli/src/cli.rs:1131]: Hmm, you have `derived-address-program-id` Arg requires `derived-address-seed`, but not the other way around. So I was expecting `derived_address_program_id` to default to something (maybe the system program?) if only `derived_address_seed.is_some()`. Could be useful, although I'm not sure.
If your intent is both are required if either is provided, let's set that in the clap Args.

**Review by CriesofCarrots — DISMISSED** [review]
Nice how low-profile adding this support turned out to be. r+ couple nits

File: cli/src/cli.rs (hunk 2 above)
Comment (t-nelson) [reply cli/src/cli.rs:1131]: Oh!  For some reason I was under the impression that setting one requirement linked them. I'll add the requirement

**Pull request merged**